### PR TITLE
FQN Removed and Make the FQName comparable.

### PR DESCRIPTION
**Pull request by piyush-gupta1999**
# Terms

> THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE TERMS OF THE CCLA DATED 2017-11-07 WITH FINOS/LINUX FOUNDATION (FORMERLY THE SYMPHONY SOFTWARE FOUNDATION CCLA).
>
> THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

**Comment by piyush-gupta1999** [comment]
Since I pulled the updated code. I got an error in the SpringBoot Backend file. Someone has pushed the backend file with errors. Please fix it so that no one got an error during the build.

File: src/Morphir/Type/MetaTypeMapping.elm [hunk 1]
```diff
@@ -35,12 +35,15 @@ lookupConstructor baseVar refs ((FQName packageName moduleName localName) as fQN
...
+                                        |> List.filterMap
+                                            (\(Type.Constructor ctorName ctorArgs) ->
+                                                if ctorName == localName then
+                                                    Just (ctorToMetaType baseVar refs (MetaRef ( packageName, moduleName, typeName )) paramNames (ctorArgs |> List.map Tuple.second))
 
-                                        Nothing ->
-                                            []
+                                                else
+                                                    Nothing
+                                            )
```
Comment (AttilaMihaly) [inline src/Morphir/Type/MetaTypeMapping.elm:47]: There seems to be a merge issue here. You are trying to overwrite my changes which is why the build fails.

File: src/Morphir/SpringBoot/Backend.elm [hunk 2]
```diff
@@ -208,19 +207,20 @@ mapStatefulAppImplementation opt distribution currentPackagePath currentModulePa
...
+                                                        |> List.concatMap
+                                                            (\( _, consType ) ->
+                                                                case consType of
+                                                                    Type.Reference _ ( _, _, consTypeName ) _ ->
+                                                                        [ consTypeName |> Name.toTitleCase ]
+
+                                                                    _ ->
+                                                                        []
+                                                            )
                                         )
```
Comment (AttilaMihaly) [inline src/Morphir/SpringBoot/Backend.elm:224]: Same merge issue here. You should keep my code without the `Type.Constructor` which I removed.

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]

**Comment by AttilaMihaly** [comment]
> Since I pulled the updated code. I got an error in the SpringBoot Backend file. Someone has pushed the backend file with errors. Please fix it so that no one got an error during the build.

The chances of that happening are very low because the build checks run for every PR so very difficult to put in code that doesn't compile. In this case you were trying to overwrite the latest with some code that was old. Probably a merge issue. I highlighted the specific places above.

**Review by AttilaMihaly — APPROVED** [review]

**Pull request merged**